autosklearn/metrics/__init__.py

```diff
@@ -341,19 +341,22 @@ def calculate_score(
     prediction: np.ndarray,
     task_type: int,
     metric: Scorer,
-    all_scoring_functions: bool = False
+    scoring_functions: list = None
 ) -> Union[float, Dict[str, float]]:
     if task_type not in TASK_TYPES:
         raise NotImplementedError(task_type)
 
-    if all_scoring_functions:
+    if scoring_functions:
         score_dict = dict()
         if task_type in REGRESSION_TASKS:
             # TODO put this into the regression metric itself
             cprediction = sanitize_array(prediction)
             metric_dict = copy.copy(REGRESSION_METRICS)
             metric_dict[metric.name] = metric
             for metric_ in REGRESSION_METRICS:
+                if metric_ not in scoring_functions:
```

Comment:
Wouldn't it make more sense to loop through the scoring functions and apply all which are in there? This would allow custom scoring functions, while this change doesn't.

Reply:
Makes sense. Made required changes.